SCClassLibrary/Common/Audio/BufIO.sc

```diff
@@ -137,7 +137,7 @@ Tap : UGen {
 	*ar { arg bufnum = 0, numChannels = 1, delaytime = 0.2;
 		var n;
 		n = delaytime * SampleRate.ir.neg; // this depends on the session sample rate, not buffer.
-		^PlayBuf.ar(numChannels, bufnum, 1, 0, n, 1);
+		^PlayBuf.ar(numChannels, bufnum, BufRateScale.kr(bufnum) * 1, 0, n, 1);
```

Comment:
Multiplication by 1 is a no-op -- so then it takes a few CPU cycles to start to create the `*` operator, but then discard it as a degenerate case. For cleanliness, just omit.

Then -- at this point, if line 139 uses BufRateScale, it will be inefficient to have two separate BufRateScale units. So you should really do:

```
		var n;
		var scale = BufRateScale.kr(bufnum);
		n = delaytime * (SampleRate.ir.neg * scale);
		^PlayBuf.ar(numChannels, bufnum, scale, 0, n, 1);
```

Reply:
Thanks for neat code suggestion and kind explain.
I apply this code as it is. Best :)